packages/block-library/src/post-comments/index.php

```diff
@@ -66,3 +66,20 @@ function register_block_core_post_comments() {
 	);
 }
 add_action( 'init', 'register_block_core_post_comments' );
+
+if ( ! function_exists( 'wp_post_comments_block_form_defaults' ) ) {
+	/**
+	 * Use the button block classes for the form-submit button.
+	 *
+	 * @param array $fields The default comment form arguments.
+	 *
+	 * @return array Returns the modified fields.
+	 */
+	function wp_post_comments_block_form_defaults( $fields ) {
+		$fields['submit_button'] = '<input name="%1$s" type="submit" id="%2$s" class="%3$s wp-block-button__link" value="%4$s" />';
+		$fields['submit_field']  = '<p class="form-submit wp-block-button">%1$s %2$s</p>';
+
+		return $fields;
+	}
+	add_filter( 'comment_form_defaults', 'wp_post_comments_block_form_defaults' );
```

Comment:
In the Gutenberg plugin where we have both Post Comments and Post Comments Form, this filter will have two functions which do the same thing hooked into it. Is that okay?

Reply:
Yes, that's OK. The filter _replaces_ the button HTML to add the necessary class, and in both cases the result is identical so there are no side-effects like duplicate classes etc. 
Ideally when back porting to WP this should be implemented directly in the `comment_form` function 👍